test/unit/helpers/form/ElementMapFactoryTest.php

```diff
@@ -0,0 +1,509 @@
+                ]
+            );
+
+        $valueCollection = $this->createMock(ValueCollection::class);
+
+        $this->valueCollectionService
+            ->method('findAll')
+            ->withAnyParameters()
+            ->willReturn($valueCollection);
+
+        $this->ruleRegistry
+            ->method('getValidators')
+            ->with($property)
+            ->willReturn($validators);
+
+        if ($instance !== null) {
+            $this->sut->withInstance($instance);
+        }
+
+        $this->sut->withStandaloneMode($standalone);
+        $this->sut->withElement($elementForWidget);
+        $this->sut->withValueCollectionService($this->valueCollectionService);
+
+        $elementForWidget
+            ->expects($this->atLeastOnce())
+            ->method('setDescription')
+            ->with($expected->getDescription());
+
+        if ($elementForWidget instanceof tao_helpers_form_elements_MultipleElement) {
+            $elementForWidget
+                ->expects($this->atLeastOnce())
+                ->method('setOptions')
+                ->with($expectedOptions);
+        }
+
+        if ($listDependencyEnabled) {
+            $dependencyCollection = $this->createMock(
+                DependsOnPropertyCollection::class
+            );
+
+            $parentProperty = $this->createMock(
+                core_kernel_classes_Property::class
+            );
+
+            $parentProperty
+                ->method('getUri')
+                ->willReturn('url://parent');
+
+            $dependencyCollection
+                ->method('offsetExists')
+                ->with(0)
+                ->willReturn(true);
+
+            $dependencyCollection
+                ->method('offsetGet')
+                ->with(0)
+                ->willReturn($parentProperty);
+
+            $property
+                ->expects($this->once())
+                ->method('getDependsOnPropertyCollection')
+                ->with()
+                ->willReturn($dependencyCollection);
+
+            $elementForWidget
+                ->expects($this->atLeastOnce())
+                ->method('addAttribute')
+                ->with('data-depends-on-property', 'url://parent');
+
+            $instance
+                ->method('getPropertyValuesCollection')
+                ->with($parentProperty)
+                ->willReturn([
+                    'parentPropertyValue1',
+                    'parentPropertyValue2',
+                ]);
+        }
+
+        $element = $this->sut->create($property);
+
+        $this->assertInstanceOf(tao_helpers_form_FormElement::class, $element);
+    }
+
+    public function successScenariosDataProvider(): array
+    {
+        return [
+            'Happy Path' => [
+                'expected' => $this->mockNamedElement(
+                    tao_helpers_form_elements_Authoring::WIDGET_ID,
+                    'property label'
+                ),
+                'expectedOptions' => [
+
+                ],
```

Comment:
As I see it, in all cases the `expectOptions` element is empty - do we need it?
```suggestion
                'expectedOptions' => [],
```

Reply:
It is not needed _yet_: The idea is to use this variable to pass the options `ElementMapFactory::create()` should have set in the `tao_helpers_form_FormElement` instance it returns, but that is not being tested yet as part of this PR to prevent making it bigger.